pkg/streamingpromql/planning/plan.go

```diff
@@ -261,8 +271,12 @@ func NodeTypeName(n Node) string {
 // ToDecodedPlan converts this encoded plan to its decoded form.
 // It returns references to the specified nodeIndices.
 func (p *EncodedQueryPlan) ToDecodedPlan(nodeIndices ...int64) (*QueryPlan, []Node, error) {
+	if p.Version > MaximumSupportedQueryPlanVersion {
+		return nil, nil, apierror.Newf(apierror.TypeBadData, "query plan has version %v, but the maximum supported query plan version is %v", p.Version, MaximumSupportedQueryPlanVersion)
```

Comment:
Why the use of `apierror` here? It seems odd to use an error specific to the Prometheus HTTP API.

Reply:
Maybe I've been using `apierror` wrong this whole time - I view it as a way to hint what kind of error it is, so that the API endpoint logic doesn't need to know about every kind of error and how to translate that to an error code / HTTP status code.

Is there something else you had in mind instead?